color/src/colorspace.rs

```diff
@@ -58,6 +58,21 @@ pub trait Colorspace: Clone + Copy + 'static {
         let scaled = LinearSrgb::scale_chroma(rgb, scale);
         Self::from_linear_srgb(scaled)
     }
+
+    /// Clip the color's components to the range allowed by the colorspace.
+    ///
+    /// The resultant color is guaranteed to be inside the bounds (and thus gamut) of the
```

Comment:
I think what I'd say here is that there are many possible mappings from out-of-gamut colors to the gamut boundary. Some may be perceptually better than others, for example preserving hue. This method will generally do the mathematically simplest thing, namely clamping of individual component values.